platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -172,9 +172,17 @@ class ConfigurationCacheProblems(
 
     private
     fun reportIncompatibleTask(trace: PropertyTrace, reason: String) {
+
         val problem = problemFactory
             .problem {
-                text(trace.containingUserCode)
+                if (trace is PropertyTrace.Task) {
+                    text("task ")
+                    reference(trace.path)
+                    text(" of type ")
+                    reference(trace.type)
```

Comment:
💭  This seems to repeat part of the logic of `containingUserCode`, but instead of a string the `StructuredMessage` is built. Should we implement `trace.containingUserCodeMessage` on the `PropertyTrace` side to cover all cases reliably?

Reply:
👍 Yes, let's expose the underlying structured message via another property and use that one instead here.

Reply:
Reworking this so we can share the description logic for property traces nd use it in both contexts: console and report. Will be sharing the reworked version tomorrow.

Reply:
@alllex  PTA(nother)L